image_data_sar/sched.py

```diff
@@ -0,0 +1,262 @@
+                           for roll_option in aca.roll_options):
+                        break
+
+                aca.sort_and_limit_roll_options(roll_level, max_roll_options)
+
+            except Exception:  # as err:
+                err = traceback.format_exc()
+                aca.add_message('critical', text=f'Running get_roll_options() failed: \n{err}')
+                aca.roll_options = None
+                aca.roll_info = None
+
+        if make_html:
+
+            # Output directory for the main prelim review index.html and for this obsid.
+            # Note that the obs{aca.obsid} is not flexible because it must match the
+            # convention used in ACATable.make_report().  Oops.
+            aca.preview_dir = Path(report_dir)
+            aca.obsid_dir = aca.preview_dir / f'obs{aca.obsid}'
+            aca.obsid_dir.mkdir(parents=True, exist_ok=True)
+
+            aca.make_starcat_plot()
+
+            if report_level == 'all' or aca.messages >= report_level:
+                try:
+                    aca.make_report()
+                except Exception:
+                    err = traceback.format_exc()
+                    aca.add_message('critical', text=f'Running make_report() failed:\n{err}')
+
+            if aca.roll_info:
+                aca.make_roll_options_report()
+
+            aca.context['text_pre'] = aca.get_text_pre()
+            aca.context['call_args'] = aca.get_call_args_pre()
+
+    # noinspection PyDictCreation
+    if make_html:
+
+        # Create new context or else use a copy of the supplied dict
+        if context is None:
+            context = {}
+        else:
+            context = context.copy()
+
+        context['load_name'] = load_name.upper()
+        context['proseco_version'] = proseco.__version__
+        context['sparkles_version'] = sparkles.__version__
+        context['chandra_aca_version'] = chandra_aca.__version__
+        context['acas'] = acars
+        context['summary_text'] = get_summary_text(acars)
+
+        # Special case when running a set of roll options for one obsid
+        is_roll_report = all(aca.is_roll_option for aca in acars)
+
+        label_frame = 'ACA' if aca.is_ER else 'Target'
+        context['id_label'] = f'{label_frame} roll' if is_roll_report else 'Obsid'
+
+        template_file = SPARKLES_DIR / 'index_template_preview.html'
+        template = Template(open(template_file, 'r').read())
+        out_html = template.render(context)
+
+        out_filename = report_dir / 'index.html'
+        if loud:
+            print(f'Writing output review file {out_filename}')
+        with open(out_filename, 'w') as fh:
+            fh.write(out_html)
+
+        if open_html:
+            import webbrowser
+            out_url = f'file://{out_filename.absolute()}'
+            print(f'Open URL in browser: {out_url}')
+            webbrowser.open(out_url)
+
+
+def sub_in_stars(aca):
+
+    # Sort the candidates by mag
+    aca.guides.cand_guides.sort('mag')
```

Comment:
Yeah, I was admittedly lazy (but did recognize this was an in-place operation, on an object that isn't really used again with an original sort that also isn't used for anything).  And sure, I can walk the offset lists instead of taking them out in off-by-one order, to improve readability.